### PR TITLE
add yolov4 model

### PR DESCRIPTION
**Pull request by yuedongli1**
Thank you for your contribution to the MindYOLO repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindyolo/blob/master/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

add yolov4 model

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: configs/yolov4/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# YOLOv4
```
Comment (zhanghuiyao) [inline configs/yolov4/README.md:1]: 外层modelzoo.md也加上yolov4

File: configs/yolov4/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+<img src="https://github.com/yuedongli1/images/raw/master/mindyolo20230509.png"/>
+</div>
+
+## Results
+
+<div align="center">
+
+| Name   | Scale        | Context  | ImageSize | Dataset      | Box mAP (%) | Params | FLOPs | Recipe                                                                                        | Download                                                                                                      |
+|--------|--------------|----------|-----------|--------------|-------------|--------|-------|-----------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------|
+| YOLOv4 | CSPDarknet53 | D910x8-G | 608       | MS COCO 2017 | 45.8        | 27.6M  | 52G   | [yaml]      | [weights]  |
```
Comment (zhanghuiyao) [inline configs/yolov4/README.md:32]: 权重路径和yaml路径补充一下

File: configs/yolov4/README.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+<details open>
+
+#### - Distributed Training
+
+It is easy to reproduce the reported results with the pre-defined training recipe. For distributed training on multiple Ascend 910 devices, please run
+```shell
+# distributed training on multiple GPU/Ascend devices
+mpirun -n 8 python train.py --config ./configs/yolov3/yolov3.yaml --device_target Ascend --is_parallel True --epochs 320
+```
```
Comment (zhanghuiyao) [inline configs/yolov4/README.md:56]: v3->v4，整个pr都检查下

File: configs/yolov4/README.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+**Note:**  As the global batch size  (batch_size x num_devices) is an important hyper-parameter, it is recommended to keep the global batch size unchanged for reproduction or adjust the learning rate linearly to a new global batch size.
+
+#### - Standalone Training
+
+If you want to train or finetune the model on a smaller dataset without distributed training, please run:
+
+```shell
+# standalone training on a CPU/GPU/Ascend device
+python train.py --config ./configs/yolov3/yolov3.yaml --device_target Ascend --epochs 320
```
Comment (zhanghuiyao) [inline configs/yolov4/README.md:71]: 同上

File: MODEL_ZOO.md [hunk 5]
```diff
@@ -1,16 +1,17 @@
...
+| YOLOv5 | L            | D910x8-G | 640       | MS COCO 2017 | 48.5        | 46.5M   | 109.1G | [yaml](https://github.com/mindspore-lab/mindyolo/blob/master/configs/yolov5/yolov5l.yaml)     | [weights](https://download.mindspore.cn/toolkits/mindyolo/yolov5/yolov5l_300e_mAP485-a28bce73.ckpt)          |
+| YOLOv5 | X            | D910x8-G | 640       | MS COCO 2017 | 50.5        | 86.7M   | 205.7G | [yaml](https://github.com/mindspore-lab/mindyolo/blob/master/configs/yolov5/yolov5x.yaml)     | [weights](https://download.mindspore.cn/toolkits/mindyolo/yolov5/yolov5x_300e_mAP505-97d36ddc.ckpt)          |
+| YOLOv4 | CSPDarknet53 | D910x8-G | 608       | MS COCO 2017 | 45.8        | 27.6M | 52G    | [yaml](https://github.com/mindspore-lab/mindyolo/blob/master/configs/yolov4/yolov4.yaml)      | [weights](https://download.mindspore.cn/toolkits/mindyolo/yolov4/yolov4-cspdarknet53_320e_map458-bdfc3205.ckpt) |
```
Comment (zhanghuiyao) [inline MODEL_ZOO.md:13]: yaml、weight、Scale做修改为silu版本规格

**Review by zhanghuiyao — APPROVED** [review]

File: configs/yolov4/README.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+- Context: Training context denoted as {device}x{pieces}-{MS mode}, where mindspore mode can be G - graph mode or F - pynative mode with ms function. For example, D910x8-G is for training on 8 pieces of Ascend 910 NPU using graph mode.
+- Box mAP: Accuracy reported on the validation set.
+
+## Quick Start
+
+Please refer to the [GETTING_STARTED](https://github.com/mindspore-lab/mindyolo/blob/master/GETTING_STARTED.md) in MindYOLO for details.
+
+### Training
+
+<details open>
```
Comment (CaitinZhao) [inline configs/yolov4/README.md:48]: 写一下预训练模型的获取和转换

**Review by zhanghuiyao — APPROVED** [review]

**Review by CaitinZhao — APPROVED** [review]
